packages/devtools_app/lib/src/screens/deep_link_validation/project_root_selection/select_project_view.dart

```diff
@@ -59,25 +60,47 @@ class _SelectProjectViewState extends State<SelectProjectView>
     });
   }
 
-  void _handleValidateProject(String directory) async {
-    setState(() {
-      _retrievingFlutterProject = true;
-    });
+  Future<List<String>> _requestAndridVariants(String directory) async {
     ga.timeStart(gac.deeplink, gac.AnalyzeFlutterProject.loadVariants.name);
     final List<String> androidVariants =
         await server.requestAndroidBuildVariants(directory);
-    if (!mounted) {
+    if (androidVariants.isEmpty || !mounted) {
+      // If the project is not a Flutter project, cancel timing and return an empty list.
       ga.cancelTimingOperation(
         gac.deeplink,
         gac.AnalyzeFlutterProject.loadVariants.name,
       );
+      return [];
+    } else {
+      ga.timeEnd(gac.deeplink, gac.AnalyzeFlutterProject.loadVariants.name);
+      return androidVariants;
+    }
+  }
+
+  Future<XcodeBuildOptions> _requestiOSBuildOptions(String directory) async {
+    ga.timeStart(
+      gac.deeplink,
+      gac.AnalyzeFlutterProject.loadIosBuildOptions.name,
+    );
+    final XcodeBuildOptions iosBuildOptions =
+        await server.requestIosBuildOptions(directory);
+    ga.timeEnd(
+      gac.deeplink,
+      gac.AnalyzeFlutterProject.loadIosBuildOptions.name,
+    );
+    return iosBuildOptions;
+  }
+
+  void _handleValidateProject(String directory) async {
+    setState(() {
+      _retrievingFlutterProject = true;
+    });
+    final List<String> androidVariants =
```

Comment:
before we request android variants and ios build options, I would recommend checking if this is a flutter app directly instead of inferring this value based on the set of android build variants returned from the server.

```
final connected = serviceConnection.serviceManager.connectedState.value.connected;
if (connected && !serviceConnection.serviceManager.connectedApp!.isFlutterAppNow) {
  // show dialog for non-flutter project
  return;
}
// then check android build variants, which will still need to
// handle the case where the build variants come back empty
// which could mean that a non-flutter project was selected when
// we do not have a live app connection
```